dexec/cmd.go

```diff
@@ -80,8 +80,13 @@ type Cmd struct {
 	*exec.Cmd
 	osCancel context.CancelFunc
 
+	// DisableLogging disables all logging.
 	DisableLogging bool
 
+	// DisableIOLogging disables logging of data sent over stdin, stdout, and stderr; but still
+	// logs information about the process launch and termination
+	DisableIOLogging bool
```

Comment:
Why not use an integer instead of several booleans? Example:

```
type LoggingFlag int
const (
  LogExec       = LoggingFlag(1)
  LogSignals    = LoggingFlag(2)
  LogStdIn      = LoggingFlag(4)
  LogStdOut     = LoggingFlag(8)
  LogStdErr     = LoggingFlag(16)
  LogEverything = LogExec | LogSignals | LogStdIn | LogStdOut | LogStdErr
)

Another improvement would be if it was possible to set the log-level that is used.

Reply:
It would be even better if this setting could be added to the `context.Context` for the `dexec.Command` to pick up, unless explicitly provided. That way, it could be used for a group of commands.

Reply:
I'm dropping that commit. That's the part of this PR that I care about the least, I threw it in because it seemed easy, if we want a more complex solution, we can discuss that later.

PS, that snippet would be better written using `iota`.

```go
type LoggingFlag int

const (
	LogExec = LoggingFlag(1 << iota)
	LogSignals
	LogStdIn
	LogStdOut
	LogStdErr
	LogEverything = LogExec | LogSignals | LogStdIn | LogStdOut | LogStdErr
)
```

Reply:
The commit I dropped is now available on the `lukeshu/no-io-log` branch, if we ever want to get it back.

Reply:
Oh, didn't know `iota` could be used like that. Nice.